core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -183,26 +220,53 @@ public void run() {
     // must be synchronized as it update schedulableNextDate and schedulable, and will be executed on the flow listener thread
     private synchronized void computeSchedulable(List<Flow> flows) {
         this.schedulableNextDate = new HashMap<>();
-
+        if (!this.started) {
+            triggerState.findAllForAllTenants().forEach(trigger -> {
+                triggerStateSaved.put(trigger.uid(), trigger);
+            });
+        }
+        this.started = true;
         this.schedulable = flows
             .stream()
             .filter(flow -> flow.getTriggers() != null && !flow.getTriggers().isEmpty())
             .filter(flow -> !flow.isDisabled() && !(flow instanceof FlowWithException))
             .flatMap(flow -> flow.getTriggers()
                 .stream()
                 .filter(abstractTrigger -> !abstractTrigger.isDisabled() && abstractTrigger instanceof PollingTriggerInterface)
-                .map(trigger -> {
-                    RunContext runContext = runContextFactory.of(flow, trigger);
-
-                    return new FlowWithTrigger(
+                .map(abstractTrigger -> {
+                    RunContext runContext = runContextFactory.of(flow, abstractTrigger);
+                    ConditionContext conditionContext = conditionService.conditionContext(runContext, flow, null);
+                    Trigger triggerContext = null;
+                    try {
+                        Trigger lastTriggers = triggerStateSaved.get(Trigger.uid(flow, abstractTrigger));
+                        if (lastTriggers != null && lastTriggers.getNextExecutionDate() == null) {
+                            triggerContext = Trigger.builder()
+                                .tenantId(flow.getTenantId())
+                                .namespace(flow.getNamespace())
+                                .flowId(flow.getId())
+                                .flowRevision(flow.getRevision())
+                                .triggerId(abstractTrigger.getId())
+                                .date(now())
+                                .nextExecutionDate(((PollingTriggerInterface) abstractTrigger).nextEvaluationDate(conditionContext, Optional.empty()))
+                                .build();
+                            triggerState.save(triggerContext);
+                        } else {
+                            triggerContext = lastTriggers;
+                        }
+                    } catch (Exception e) {
+                        logError(conditionContext, flow, abstractTrigger, e);
```

Comment:
I think you should return null if there is an issue and filter on null just after